data/admin/vegetarianexpe.py

```diff
@@ -0,0 +1,129 @@
+from django.contrib import admin
+from data.models import VegetarianExpe
+
+
+class VegetarianExpeParticipantFilter(admin.SimpleListFilter):
+    title = "Participant à l'experimentation"
+    parameter_name = "canteen.vegetarian_expe_participant"
+
+    def lookups(self, request, model_admin):
+        return (
+            (None, "Participants"),
+            ("non-participants", "Ne participent pas"),
+            ("all", "Tous"),
+        )
+
+    def choices(self, cl):
+        for lookup, title in self.lookup_choices:
+            yield {
+                "selected": self.value() == lookup,
+                "query_string": cl.get_query_string(
+                    {
+                        self.parameter_name: lookup,
+                    },
+                    [],
+                ),
+                "display": title,
+            }
+
+    def queryset(self, request, queryset):
+        if self.value() is None:
+            return queryset.filter(canteen__reservation_expe_participant=True)
+        elif self.value() in ("all"):
+            return queryset
+        elif self.value() in ("non-participants"):
+            return queryset.filter(canteen__reservation_expe_participant=False)
```

Comment:
I assume these if statements should use `vegetarian_expe_participant`

Reply:
Absolutely : https://github.com/betagouv/ma-cantine/pull/1237/commits/c757fed5ea875c0a60f26bd6394832dd07801bc9